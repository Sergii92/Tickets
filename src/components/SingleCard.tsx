import React from "react";

import { Card } from "react-bootstrap";

import { CustomCard } from "../styled/components";

interface CardProps {
  id: number;
  sessionName: string;
  time: string;
  onClick: () => void;
}

const SingleCard = ({ id, sessionName, time, onClick }: CardProps) => {
  return (
    <>
      <CustomCard className="text-center" onClick={onClick}>
        <Card.Body>
          <Card.Title>{sessionName}</Card.Title>
          <Card.Text>Open For get more Information</Card.Text>
          <Card.Text>
            <small className="text-muted">Session Time:{time}</small>
          </Card.Text>
        </Card.Body>
      </CustomCard>
    </>
  );
};

export default SingleCard;
