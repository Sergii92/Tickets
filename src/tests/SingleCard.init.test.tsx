import React from "react";
import { shallow, render } from "enzyme";

import SingleCard from "../components/SingleCard";

describe("SingleCard Component", () => {
  const props = {
    daysId: 1,
    time: "10:00",
    id: 11,
    description:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam possimus quaerat ipsa dolor nihil cum ullam iste culpa veniam, magni a neque beatae consequatur totam eaque maxime aperiam illum molestiae.",
    sessionName: "Sunday session1",
    onClick: () => null,
  };

  it("RENDER", () => {
    const component = shallow(<SingleCard {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("find title", () => {
    const component = render(<SingleCard {...props} />);
    expect(component.find(".card-title").text()).toEqual(props.sessionName);
  });
  it("find text", () => {
    const component = render(<SingleCard {...props} />);
    expect(component.find(".card-text").first().text()).toEqual(
      "Open For get more Information"
    );
  });
});
