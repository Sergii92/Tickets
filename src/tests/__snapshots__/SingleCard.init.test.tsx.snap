// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCard Component RENDER 1`] = `ShallowWrapper {}`;
